src/main/java/org/keybase/NotaryApp.java

```
@@ -31,231 +32,270 @@
+		    System.out.println("No new Merkle Roots to issue");
+		waitTime();
+	    }
+	} catch(Throwable e) {
+	    System.err.println("Exception occurred: " + Throwables.getStackTraceAsString(e));
+	}
     }
 
+    /** Waits secondsBetween seconds. This is called inbetween
+     * attempts to issue a new Merkle Root.
+     */
     private static void waitTime() {
-    	try {
-    		Thread.sleep(1000 * secondsBetween);
-    	} catch (InterruptedException e) {
-    		Thread.currentThread().interrupt();
-    	}
+	try {
+	    Thread.sleep(1000 * secondsBetween);
+	} catch (InterruptedException e) {	    
+	    Thread.currentThread().interrupt();
+	}
     }
 
     private static void issueStatementHandler(int sequenceNumber) throws InsufficientMoneyException, IOException, InterruptedException {
-        // If there's no root-of-trust TXN, we gotta take care of that first
-        if(ext.hasRootOfTrustTxid() == false) {
-            issueRootOfTrustHandler();
-            return;
-        } else {
-            log.debug("Already have root-of-trust txid '{}'", ext.getRootOfTrustTxid());
-        }
-
-        String stmt;
-        try {
-        	stmt = JSONParser.getMerkleRoot(sequenceNumber);
-        } catch (Exception e) {
-        	stmt = "";
-        	System.exit(1);
-        }
-        //        System.out.print("Please type in your next issued statement: ");
-        //        String stmt = scanner.nextLine();
-        //        if(stmt.trim().isEmpty()) {
-        //            System.out.println("Cancelled due to empty statement. Please try again but type something in.");
-        //            return;
-        //        }
+	// If there's no root-of-trust TXN, we gotta take care of that first
+	if(ext.hasRootOfTrustTxid() == false) {
+	    issueRootOfTrustHandler();
+	    //return;
+	} else {
+	    log.debug("Already have root-of-trust txid '{}'", ext.getRootOfTrustTxid());
+	}
+
+	String stmt;
+	try {
+	    stmt = JSONParser.getMerkleRoot(sequenceNumber);
```

So because you swallowed the exception in getMerkleRoot, you could have an empty stmt "" returned here, which you forgot to handle. Now you might call issueStatement with that empty stmt, and issueStatement does not handle it either (and it shouldn't anyway: we wanna maintain an invariant about what we give as input to issueStatement  so that it can be _less complex_).

Anyway, here you should catch an exception from getMerkleRoot and handle it accordingly. If it's a bad URL, then the daemon should log a message using log.error() and exit. If it's a network I/O error then you should retry with exponential back off up until a reasonable interval of 5/10 mins or something.